ts_src/psetv2/updater.ts

```diff
@@ -54,12 +78,50 @@ export class Updater {
     this.pset = pset;
   }
 
-  addInputs(ins: CreatorInput[]): this {
+  addInputs(ins: UpdaterInput[]): this {
     const pset = this.pset.copy();
 
     ins.forEach(input => {
-      input.validate();
-      pset.addInput(input.toPartialInput());
+      const creatorInput = new CreatorInput(
+        input.txid,
+        input.txIndex,
+        input.sequence,
+        input.heightLocktime,
+        input.timeLocktime,
+      );
+      creatorInput.validate();
+      pset.addInput(creatorInput.toPartialInput());
+
+      // we know at this point, index can't be negative
+      const inputIndex = pset.inputs.length - 1;
+
+      if (input.witnessUtxo)
+        this.addInWitnessUtxo(inputIndex, input.witnessUtxo);
+
+      if (input.witnessUtxo && input.witnessUtxo.rangeProof) {
+        this.addInUtxoRangeProof(inputIndex, input.witnessUtxo.rangeProof);
+      }
+
+      if (input.nonWitnessUtxo)
+        this.addInNonWitnessUtxo(inputIndex, input.nonWitnessUtxo);
```

Comment:
as done for the witnessUtxo, we should try to set the input utxo rangeproof in case the nonWitness prevout is confidential.

Reply:
fixed https://github.com/vulpemventures/liquidjs-lib/pull/68/commits/62b3bf416e45c75e01c8ca07db988e7c0977ce06